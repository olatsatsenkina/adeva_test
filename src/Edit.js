import React from 'react';

export default class Edit extends React.Component {
  constructor(props) {
    super(props);
    //this "magic" is required because localStorage uses commas to separate array's values
    let value = this.props.text && this.props.text.replace(/&#44;/g, ",");
    this.state = {
      value: value
    }
    this.onSave = this.onSave.bind(this);
  }

  changeHandler(el){
    let value = el.currentTarget.value;
    this.setState({value: value});
  }

  componentWillReceiveProps(props) {
    //this "magic" is required because localStorage uses commas to separate array's values
    let value = props.text && props.text.replace(/&#44;/g, ",")
    this.setState({value: props.text})
  }

  onSave() {
    this.props.onSave(this.state.value);
  }

  onDelete() {
    this.props.onDelete();
  }

  render() {
    let {mode} = this.props;
    let title = (mode == 'add') ? 'Add note' : 'Edit your note';
    let deleteCancel = (mode == 'add') ? 'Cancel' : 'Delete';
    let addSave = (mode == 'add') ? 'Add' : 'Save';
    return (
      <div className="dashboard">
        <h2>{ title }</h2>
        <div className="notes-contents">
            <a href="#" className="delete-note" onClick={this.onDelete.bind(this)}>{deleteCancel}</a>
            <textarea
              onChange={this.changeHandler.bind(this)}
              placeholder="Note's text"
              className="note-text md-ish"
              value={this.state.value}>
            </textarea>
            <button onClick={this.onSave.bind(this)} className="save-note md-ish hover-cursor">{addSave}</button>
        </div>
      </div>
    )
  }
}
