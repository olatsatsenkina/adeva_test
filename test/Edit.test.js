import React from 'react';
import Edit from '../src/Edit.js';
import renderer from 'react-test-renderer';

it('Edit interface renders correctly', () => {
  const tree = renderer
    .create(<Edit
      mode="edit"
      text="Some test text here"/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Add interface renders correctly', () => {
  const tree = renderer
    .create(<Edit
      mode="add"
      text=""/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});


let saveNoteHandler = (value) => {
  expect(value).toBe("Some text here");
}

let deleteNoteHandler = () => {
  expect(true).toBe(true);
}

test('Save in edit mode is working', () => {
  const editComponent = renderer.create(<Edit mode='edit'
    text="Some text here"
    onSave={saveNoteHandler}/>);
  const rootInstance = editComponent.root;
  const button = rootInstance.findByType("button");
  button.props.onClick();
});

test('Delete note in edit mode is working', () => {
  const editComponent = renderer.create(<Edit mode='edit'
    text="Some text here"
    onDelete={deleteNoteHandler}/>);
  const rootInstance = editComponent.root;
  const link = rootInstance.findByType("a");
  link.props.onClick();
});

let saveNoteHandlerUpdate = (value) => {
  expect(value).toBe("Updated note");
}

test('Change contents of the texarea is ok', () => {
  const editComponent = renderer.create(<Edit mode='edit'
    onSave={saveNoteHandlerUpdate}
    text="Some text here" />);
  const rootInstance = editComponent.root;
  const textarea = rootInstance.findByType("textarea");
  let e = {currentTarget: {value: "Updated note"}}
  textarea.props.onChange(e);
  const button = rootInstance.findByType("button");
  button.props.onClick();
});
