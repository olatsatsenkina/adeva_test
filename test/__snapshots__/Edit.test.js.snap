// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add interface renders correctly 1`] = `
<div
  className="dashboard"
>
  <h2>
    Add note
  </h2>
  <div
    className="notes-contents"
  >
    <a
      className="delete-note"
      href="#"
      onClick={[Function]}
    >
      Cancel
    </a>
    <textarea
      className="note-text md-ish"
      onChange={[Function]}
      placeholder="Note's text"
      value=""
    />
    <button
      className="save-note md-ish hover-cursor"
      onClick={[Function]}
    >
      Add
    </button>
  </div>
</div>
`;

exports[`Edit interface renders correctly 1`] = `
<div
  className="dashboard"
>
  <h2>
    Edit your note
  </h2>
  <div
    className="notes-contents"
  >
    <a
      className="delete-note"
      href="#"
      onClick={[Function]}
    >
      Delete
    </a>
    <textarea
      className="note-text md-ish"
      onChange={[Function]}
      placeholder="Note's text"
      value="Some test text here"
    />
    <button
      className="save-note md-ish hover-cursor"
      onClick={[Function]}
    >
      Save
    </button>
  </div>
</div>
`;
